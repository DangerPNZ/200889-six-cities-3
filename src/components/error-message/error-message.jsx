import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessageComponent = ({errorData, onErrorClose}) => (
  <section className="error-message">
    <div className="error-message__header">
      <div className="error-message__header-text">Error: {errorData.heading}</div>
      <div className="error-message__close-btn" onClick={onErrorClose}></div>
    </div>
    <div className="error-message__inner-container">
      <div className="error-message__error-text">{errorData.description}</div>
    </div>
  </section>
);

ErrorMessageComponent.propTypes = {
  errorData: PropTypes.exact({
    heading: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired
  }).isRequired,

  onErrorClose: PropTypes.func.isRequired
};

export const ErrorMessage = React.memo(ErrorMessageComponent);
