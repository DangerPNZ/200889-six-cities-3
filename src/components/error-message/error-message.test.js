import React from 'react';
import renderer from 'react-test-renderer';
import {ErrorMessage} from './error-message.jsx';

const TestDataValue = {
  ERROR_DATA: {
    heading: `Test error. This heading`,
    description: `Test error. This discription.`
  }
};

it(`ErrorMessage component structure test`, () => {
  const tree = renderer
  .create(
      <ErrorMessage
        errorData = {TestDataValue.ERROR_DATA}
        onErrorClose = {() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
