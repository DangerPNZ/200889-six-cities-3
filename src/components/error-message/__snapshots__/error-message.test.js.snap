// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component structure test 1`] = `
<section
  className="error-message"
>
  <div
    className="error-message__header"
  >
    <div
      className="error-message__header-text"
    >
      Error: 
      Test error. This heading
    </div>
    <div
      className="error-message__close-btn"
      onClick={[Function]}
    />
  </div>
  <div
    className="error-message__inner-container"
  >
    <div
      className="error-message__error-text"
    >
      Test error. This discription.
    </div>
  </div>
</section>
`;
